src/libstore/build/substitution-goal.cc

```diff
@@ -110,18 +110,12 @@ void PathSubstitutionGoal::tryNext()
         tryNext();
         return;
     } catch (SubstituterDisabled &) {
-        if (settings.tryFallback) {
-            tryNext();
-            return;
-        }
-        throw;
+        tryNext();
+        return;
     } catch (Error & e) {
-        if (settings.tryFallback) {
-            logError(e.info());
-            tryNext();
-            return;
-        }
-        throw;
+        logError(e.info());
```

Comment:
ditto, not sure if we should log here, or log at the end if we don't end up re-throwing.